src/item.h

```diff
@@ -1797,6 +1797,21 @@ class item : public visitable<item>
         /** Helper for checking reloadability. **/
         bool is_reloadable_helper( const itype_id &ammo, bool now ) const;
 
+        enum class sizing {
+            human_sized_human_char = 0,
+            big_sized_human_char,
+            small_sized_human_char,
+            big_sized_big_char,
+            human_sized_big_char,
+            small_sized_big_char,
+            small_sized_small_char,
+            human_sized_small_char,
+            big_sized_small_char,
+            not_wearable
```

Comment:
Yuck! Why explicitly specify all possible combinations? This approach is awkward at best (a pain to maintain).

A character can be either: big, small, or normal sized.
A piece of clothing can be either: undersided, fit, oversized.

These are two natural and well normalized `enum`s.

Reply:
I started with that approach and found that it made the implementation harder to maintain. In addition, explicit enums make checks on those enums more straight forward and readable in the actual logic places. The item info method is already in great need of a refactor as is so I chose to move complexity into the enum declaration and out of the use cases.

In order to create constructive dialogue and maintain a level of respect, its a good idea to reframe from using expletives in code reviews.

Reply:
> In order to create constructive dialogue and maintain a level of respect, its a good idea to reframe from using expletives in code reviews.

I didn't mean to be rude or something. Apologies if it looked that way.

> I started with that approach and found that it made the implementation harder to maintain

Why not introduce a more explicit `enum` along with a function to calculate the difference? This would probably be both more readable and maintainable. Something like:

```c++
switch (armor_size - char_size) {
    case -2:  return armor_fitting::tiny;
    case -1:  return armor_fitting::too_small;
    case 0:   return armor_fitting::okay;
    case 1:   return armor_fitting::too_big;
    case 2:   return armor_fitting::huge;
}
```

Reply:
Im very much in favor of an enum like youve outlined and also had an iteration that had a relative size enum like that but, the issue there was in how specific the differences are between being small with variable clothes and being huge with fixed clothes. Since you cannot change huge clothes and cannot upsize small or normal clothes to a huge size, just knowing how the armor fits isnt enough information to tell the user why the armor doesnt fit from an armor sizing stand point and a character size stand point without introducing more complexity in the item info function. With an item info refactor and maybe a harder definition of clothing size vs fit, a relative enum like that may make since. 

In general, DRY is a good principle to follow but in this specific case, even though I also would like a simpler and more elegant enum, doing so just moves the complexity into the places its used. For maintainability, the info function will be modified more often than the enum of clothes or character sizes so we gain some ease with having that function be more straight forward